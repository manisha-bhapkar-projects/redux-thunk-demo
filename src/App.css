.bg-dark {
  background-color: #21212b;
}

.height-100 {
  height: 100vh;
  display: flex;
  align-items: center;
}