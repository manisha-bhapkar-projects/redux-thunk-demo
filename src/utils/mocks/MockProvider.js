import { Provider } from 'react-redux';
import store from '../../store/createstore';

export default function MockProvider({ children }) {
  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
};
