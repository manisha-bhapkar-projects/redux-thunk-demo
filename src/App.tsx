import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux'
import RadioWidget from './components/RadioWidget';
import './App.css';
import { fetchStation } from './store/actions/station-action-thunk';

const App = (props: any) => {

  useEffect(() => {
    props.dispatch(fetchStation())
  }, [])

  const getSelectedStation = (selectedStation) => {
    console.log(selectedStation);
  }
  const handlePlusButton = () => {
    console.log("Plus Action");
  }
  const handleMinusButton = () => {
    console.log("Minus Action");
  }
  const handleBackButton = () => { console.log("Back Action"); }
  const handleSwitchButton = () => { console.log("Switch Action"); }

  return (
    <div className="App">
      <div className="bg-dark height-100">
        <RadioWidget
          stationList={props.station}
          loading={props.loading}
          getSelectedStation={getSelectedStation}
          handlePlusButton={handlePlusButton}
          handleMinusButton={handleMinusButton}
          handleBackButton={handleBackButton}
          handleSwitchButton={handleSwitchButton}
        />
      </div>
    </div>
  );
}

const mapStateToProps = (state: any) => ({
  station: state.stationReducer.station_data,
  loading: state.stationReducer.loading
})

export default connect(mapStateToProps, null)(App)


