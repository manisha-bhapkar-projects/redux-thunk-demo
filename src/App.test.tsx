import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App';
import MockProvider from "./utils/mocks/MockProvider";
// const createTestProps = (props) => ({
//   stationList: MockProvider,
//   ...props
// })
describe("App component test suite", () => {
  it('it should render successfully', () => {
    // let props = createTestProps({})
    const comp = shallow(
      <MockProvider>
        <App />
      </MockProvider>
    );
    expect(toJson(comp.container)).toMatchSnapshot();
  });
});