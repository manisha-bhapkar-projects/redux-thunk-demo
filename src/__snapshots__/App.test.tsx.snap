// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test suite it should render successfully 1`] = `undefined`;
